### PR TITLE
BROKERS: Create Token Broker And Register It

### LINKED ISSUE #155: BROKERS: GenerateJWT()

### PR DESCRIPTION
**Pull request by khatanboyevelbek**
Closes #155

### CODE REVIEW

File: Tarteeb.Api/Brokers/Tokens/TokenBroker.cs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        public string GenerateJWT(User user)
+        {
+            var secretKey =
+                configuration.GetSection("Jwt").GetSection("Key").Value;
+
+            var jwtIssuer =
+                configuration.GetSection("Jwt").GetSection("Issuer").Value;
+
+            var jwtAudience =
+                configuration.GetSection("Jwt").GetSection("Audience").Value;
```
Comment (ElbekDeveloper) [inline Tarteeb.Api/Brokers/Tokens/TokenBroker.cs:32]: Bu yaxshiroq. Lekin hali aytganimdek emas. 2 ta yo'lingiz bor buni yaxshilash uchun
Birinchisi:

1. `TokenConfiguration` degan model yarating. (Key, Issuer, Audience propertilar bo'lsin)
2. [Bind](https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.configuration.configurationbinder.bind?view=dotnet-plat-ext-7.0#microsoft-extensions-configuration-configurationbinder-bind(microsoft-extensions-configuration-iconfiguration-system-object)) metodi yordamida bittada hammasini qiymatini oling va shu obyektni hamma joyda ishlating. 6 qatorni 1 qatorga tushurasiz shunda


Ikkinchi yo'li: 
1. `IConfigurationSection jwtSection = configuration.GetSection("Jwt");` qilib alohida olib qoyingda 
2. `string secretKey = jwtSection.GetValue<string>(key: "Key"); ` va hokazo qolganlarini ham shunday qiymatini olib chiqing. 
3. GetValue haqida [havola](https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.configuration.configurationbinder.getvalue?view=dotnet-plat-ext-7.0#microsoft-extensions-configuration-configurationbinder-getvalue(microsoft-extensions-configuration-iconfiguration-system-type-system-string))

File: Tarteeb.Api/Brokers/Tokens/TokenBroker.cs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    public class TokenBroker : ITokenBroker
+    {
+        private readonly IConfiguration configuration;
+
+        public TokenBroker(IConfiguration configuration) =>
+            this.configuration = configuration;
+
+        public string GenerateJWT(User user)
+        {
+            var secretKey =
```
Comment (ElbekDeveloper) [inline Tarteeb.Api/Brokers/Tokens/TokenBroker.cs:25]: o'ng tomon aniq bo'lmaganda chap tomonda tur yoziladi.

**Review by ElbekDeveloper — CHANGES_REQUESTED** [review]

**Comment by ElbekDeveloper** [comment]
Alohida PR ochish kerak emas bu uchun. `CODE RUB: Resolve Comments` commit qilsangiz yetadi

File: Tarteeb.Api/Brokers/Tokens/TokenBroker.cs (hunk 2 above)
Comment (khatanboyevelbek) [reply Tarteeb.Api/Brokers/Tokens/TokenBroker.cs]: Kamchiliklarini to'g'rilab qo'ydim ustoz

File: Tarteeb.Api/Brokers/Tokens/TokenBroker.cs (hunk 1 above)
Comment (khatanboyevelbek) [reply Tarteeb.Api/Brokers/Tokens/TokenBroker.cs]: Ustoz, to'g'rilab qo'ydim kamchiliklarini.

File: Tarteeb.Api/Brokers/Tokens/TokenBroker.cs [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+namespace Tarteeb.Api.Brokers.Tokens
+{
+    public class TokenBroker : ITokenBroker
+    {
+        private readonly TokenConfiguration tokenConfiguration;
+
+        public TokenBroker(IConfiguration configuration)
+        {
+            tokenConfiguration = new TokenConfiguration();
```
Comment (ElbekDeveloper) [inline Tarteeb.Api/Brokers/Tokens/TokenBroker.cs:23]: `tokenConfiguration` private field ligini bildirish uchun `this` ishlatamiz. 

`this.tokenConfiguration`  ga o'zgartirib chiqing

**Review by ElbekDeveloper — CHANGES_REQUESTED** [review]

**Review by ElbekDeveloper — APPROVED** [review]

File: Tarteeb.Api/Models/Tokens/TokenConfiguration.cs [hunk 4]
```diff
@@ -0,0 +1,15 @@
+﻿//=================================
+// Copyright (c) Coalition of Good-Hearted Engineers
+// Free to use to bring order in your workplace
+//=================================
+
+namespace Tarteeb.Api.Models.Tokens
+{
+    public class TokenConfiguration
+    {
+        public string? Key { get; set; }
```
Comment (ElbekDeveloper) [inline Tarteeb.Api/Models/Tokens/TokenConfiguration.cs:10]: string uchun ? kerak emas

File: Tarteeb.Api/Models/Tokens/TokenConfiguration.cs [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+        public string? Issuer { get; set; }
+        public string? Audience { get; set; }
+
```
Comment (ElbekDeveloper) [inline Tarteeb.Api/Models/Tokens/TokenConfiguration.cs:13]: Bu qator ham kerak emas

Comment (khatanboyevelbek) [reply Tarteeb.Api/Models/Tokens/TokenConfiguration.cs:13]: Aha to'griladim

**Review by ElbekDeveloper — APPROVED** [review]

**Pull request merged**